File: LLM_Handson/nodejs_langchain/langchain-chat-node/chat_v1.js
// chat_v1.js
import dotenv from 'dotenv';
import { ChatOpenAI } from "@langchain/openai";
import { HumanMessage, SystemMessage } from "@langchain/core/messages";
import readline from 'readline';

// Load environment variables
dotenv.config();

// Create readline interface for command line input
const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

function createChat() {
    return new ChatOpenAI({
        temperature: 0.7
    });
}

// Promise wrapper for readline
function askQuestion(query) {
    return new Promise((resolve) => {
        rl.question(query, resolve);
    });
}

async function main() {
    const chat = createChat();
    
    console.log("Welcome to SimpleChat! (type 'quit' to exit)");
    
    while (true) {
        const userInput = await askQuestion("\nYou: ");
        
        if (userInput.toLowerCase() === 'quit') {
            break;
        }
        
        try {
            const messages = [
                new SystemMessage("You are a helpful AI assistant."),
                new HumanMessage(userInput)
            ];
            
            const response = await chat.invoke(messages);
            console.log(`\nAssistant: ${response.content}`);
        } catch (error) {
            console.error("\nError:", error.message);
        }
    }
    
    rl.close();
}

// Add this to package.json to accomodate import: { "type": "module" }
main().catch(console.error);

File: LLM_Handson/python_langchain/langchain-chat/chat_v1.py
# chat_v1.py
import os
from dotenv import load_dotenv
from langchain_openai import ChatOpenAI  # Updated import
from langchain_core.messages import HumanMessage, SystemMessage  # Updated import

# Load OpenAI API key
load_dotenv()
# chat_v1.py
from langchain_openai import ChatOpenAI

def create_chat():
    # Specify model explicitly here
    return ChatOpenAI(
        temperature=0.7,
        model="gpt-3.5-turbo"  # Add this line to specify the model
    )
def create_chat():
    return ChatOpenAI(temperature=0.7)

def main():
    chat = create_chat()
    
    print("Welcome to SimpleChat! (type 'quit' to exit)")
    while True:
        user_input = input("\nYou: ")
        if user_input.lower() == 'quit':
            break
            
        messages = [
            SystemMessage(content="You are a helpful AI assistant always ready to give correct answer ,say you dont know if you are not sure."),
            HumanMessage(content=user_input)
        ]
        
        response = chat(messages)
        print(f"\nAssistant: {response.content}")

if __name__ == "__main__":
    main()